tb_plugin/torch_tb_profiler/profiler/node_parser.py

```diff
@@ -0,0 +1,94 @@
+from .. import utils
+from .node import (CommunicationNode, DeviceNode, OperatorNode,
+                   ProfilerStepNode, RuntimeNode)
+from .trace import EventTypes
+
+logger = utils.get_logger()
+
+CommunicationOpNameSet = ['nccl:broadcast', 'nccl:all_reduce']
+
+
+class NodeContext:
+    def __init__(self):
+        self.tid2list = {} # value is a list of OperatorNode and ProfilerStepNode. Do not include RuntimeNode
+        self.tid2zero_rt_list = {}  # value is a list of RuntimeNode with external_id=0. They will be attached to root nodes.
+        self.corrid_to_device = {}  # value is a list of DeviceNode
+        self.corrid_to_runtime = {}  # value is a RuntimeNode
```

Comment:
corrid_to_runtime is only used in this file, maybe it can change to local variable?